crates/proto/src/op/message.rs

```diff
@@ -141,29 +142,23 @@ impl Message {
     /// * `op_code` - operation of the request
     /// * `response_code` - the error code for the response
     pub fn error_msg(id: u16, op_code: OpCode, response_code: ResponseCode) -> Message {
-        let mut message: Message = Message::new();
-        message.set_message_type(MessageType::Response);
-        message.set_id(id);
-        message.set_response_code(response_code);
-        message.set_op_code(op_code);
+        let mut message = Message::new();
+        message
+            .set_message_type(MessageType::Response)
+            .set_id(id)
+            .set_response_code(response_code)
+            .set_op_code(op_code);
 
         message
     }
 
     /// Truncates a Message, this blindly removes all response fields and sets truncated to `true`
     pub fn truncate(&self) -> Self {
```

Comment:
Calling your attention to the `truncate` method, I *think* the goal of that is to copy the original `Message` minus any `additionals`/`answers`/`queries`?

Reply:
It makes sense to me. @bluejekyll does this match the intent, to your recollection?

Reply:
Yes, truncation in trust-dns is very aggressive. It works as @leshow says.